### PR TITLE
feat: add external dns for gcp

### PR DESCRIPTION
**Pull request by markandersontrocme**
### Description of your changes

- Add support for external-dns on GCP
- Add examples and tests for GCP
- created `gcp.upboundrocks.cloud` in `official-provider-testing` so that we have a domain to use for our tests.

I have:

- [ ] Read and followed Upbound's [contribution process](https://git.io/fj2m9).
- [ ] Run `make reviewable` to ensure this PR is ready for review.
- [ ] Added `backport release-x.y` labels to auto-backport this PR, as appropriate.

### CODE REVIEW

File: apis/space-init/composition.yaml [hunk 1]
```diff
@@ -436,6 +438,90 @@ spec:
...
+                toFieldPath: spec.parameters.id
+              - type: FromCompositeFieldPath
+                fromFieldPath: spec.parameters.operators.externaldns.gcp.dnsProjectName
+                toFieldPath: spec.parameters.project
+              - type: FromCompositeFieldPath
+                fromFieldPath: spec.parameters.operators.externaldns.gcp.managedZoneName
+                toFieldPath: spec.parameters.managedZone
+              - type: ToCompositeFieldPath
+                fromFieldPath: status.GKEprojectID
+                toFieldPath: spec.parameters.GKEProjectID
```
Comment (markandersontrocme) [inline apis/space-init/composition.yaml:467]: Figure out how to get the Project ID from the XCluster

File: apis/space-init/composition.yaml [hunk 2]
```diff
@@ -436,6 +438,90 @@ spec:
...
+                fromFieldPath: spec.parameters.operators.externaldns.version
+                toFieldPath: spec.forProvider.chart.version
+              - type: FromCompositeFieldPath
+                fromFieldPath: spec.parameters.operators.externaldns.gcp.managedZoneName
+                toFieldPath: spec.forProvider.values.domainFilters[0]
+              # - type: FromCompositeFieldPath
+              #   fromFieldPath: status.status.externalDNS.IRSARoleArn
+              #   toFieldPath: spec.forProvider.values.serviceAccount.annotations[eks.amazonaws.com/role-arn]
+              #   policy:
+              #     fromFieldPath: Required
```
Comment (markandersontrocme) [inline apis/space-init/composition.yaml:515]: Check if we need an annotation for the GSA on externaldns

File: apis/workload-identity/definition.yaml [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+  name: xworkloadidentitys.gcp.platform.upbound.io
+  labels:
+    provider: gcp
+spec:
+  claimNames:
+    kind: WorkloadIdentity
+    plural: workloadidentitys
+  group: gcp.platform.upbound.io
+  names:
+    kind: XWORKLOADIDENTITY
```
Comment (haarchri) [inline apis/workload-identity/definition.yaml:13]:
```suggestion
    kind: XWorkloadIdentity
```

**Comment by markandersontrocme** [comment]
/test-examples="examples/gcp-host-space.yaml"

**Comment by markandersontrocme** [comment]
/test-examples="examples/gcp-host-space.yaml"

**Comment by markandersontrocme** [comment]
/test-examples="examples/gcp-host-space.yaml"

**Comment by markandersontrocme** [comment]
/test-examples="examples/gcp-host-space.yaml"

File: apis/space-init/composition.yaml [hunk 4]
```diff
@@ -305,6 +305,8 @@ spec:
                     name: ingress-nginx
                     repository: https://kubernetes.github.io/ingress-nginx
                   set:
+                    - name: "nameOverride"
```
Comment (haarchri) [inline apis/space-init/composition.yaml:308]: this is a breaking change so we need to add an information

File: apis/workload-identity/composition.yaml [hunk 5]
```diff
@@ -0,0 +1,172 @@
...
+                type: FromCompositeFieldPath
+                transforms:
+                  - string:
+                      fmt: 'serviceAccount:%s'
+                      type: Format
+                    type: string
+
+          - name: workloadIdentitySettings
+            base:
+              apiVersion: kubernetes.crossplane.io/v1alpha1
```
Comment (haarchri) [inline apis/workload-identity/composition.yaml:137]: kubernetes.crossplane.io/v1alpha2

File: apis/space-init/composition.yaml (hunk 4 above)
Comment (markandersontrocme) [reply apis/space-init/composition.yaml:308]: I'll undo this in this PR to keep it simple

**Comment by markandersontrocme** [comment]
/test-examples="examples/gcp-host-space.yaml"

**Comment by markandersontrocme** [comment]
/test-examples="examples/gcp-host-space.yaml"

File: apis/workload-identity/definition.yaml [hunk 6]
```diff
@@ -0,0 +1,96 @@
+apiVersion: apiextensions.crossplane.io/v1
+kind: CompositeResourceDefinition
+metadata:
+  name: xworkloadidentitys.gcp.platform.upbound.io
+  labels:
+    provider: gcp
+spec:
+  claimNames:
+    kind: WorkloadIdentity
+    plural: workloadidentitys
```
Comment (haarchri) [inline apis/workload-identity/definition.yaml:10]: plural: workloadidentities

File: apis/workload-identity/definition.yaml [hunk 7]
```diff
@@ -0,0 +1,96 @@
...
+  group: gcp.platform.upbound.io
+  names:
+    kind: XWorkloadIdentity
+    plural: xworkloadidentitys
```
Comment (haarchri) [inline apis/workload-identity/definition.yaml:14]: plural: xworkloadidentities

**Comment by markandersontrocme** [comment]
/test-examples="examples/gcp-host-space.yaml"

**Review by haarchri — APPROVED** [review]

**Pull request merged**